service/service.go

```diff
@@ -46,19 +50,50 @@ func New(conf *config.Config) (*Service, error) {
 		return nil, fmt.Errorf("failed to create a new gRPC client: %w", err)
 	}
 
-	return &Service{
-		Conf:          conf,
-		OraclePrivKey: oraclePrivKey,
-		UniqueID:      hex.EncodeToString(selfEnclaveInfo.UniqueID),
-		GrpcClient:    grpcClient.(*panacea.GrpcClient),
-	}, nil
+	subscriber, err := event.NewSubscriber(conf.Panacea.WSAddr)
+	if err != nil {
+		// TODO: close grpcClient
+		return nil, fmt.Errorf("failed to init subscriber: %w", err)
+	}
+
+	svc := &Service{
+		conf:          conf,
+		oracleAccount: oracleAccount,
+		oraclePrivKey: oraclePrivKey,
+		uniqueID:      hex.EncodeToString(selfEnclaveInfo.UniqueID),
+		grpcClient:    grpcClient.(*panacea.GrpcClient),
+		subscriber:    subscriber,
+	}
+
+	if err := svc.startSubscriptions(); err != nil {
+		return nil, fmt.Errorf("failed to start subscriptions: %w", err)
+	}
```

Comment:
How about separating this `svc.startSubscriptions()`?
I think the `New` function just return a new svc. wdyt?

Reply:
Sorry. Didn't get what you mean. Could you elaborate it more?

Reply:
If you mean making the `startSubscriptions()` public, yes, we can. It would be good for next milestones as well, even though we cannot make sure yet. I will do that.

Reply:
Yes. I meant make it public and start subscription outside of the `New` function.
maybe
```go
svc, err := service.New(conf)
if err != nil {
    return fmt.Errorf("failed to create service: %w", err)
}

if err := svc.StartSubscription(); err != nil {
    return err
}
```

I understood. Thanks!

Reply:
https://github.com/medibloc/panacea-doracle/pull/37/commits/1f864c0e441092bc87b837df22db46e931b6b597 Done

Reply:
I made the `service.StartSubscriptions()` as general as possible, but not sure if it would be helpful for next milestones. Let's see later. Maybe, we can make it better when the next specs are defined clearly.